### PR TITLE
TECH-84: [wp-conditional-blocks] Basic Conditions

### LINKED ISSUE #18: Add some basic conditions to the plugin
### Description

We should ship the plugin with some basic conditions included.

### Use Case

there should be some functionality without any customizations

### Acceptance Criteria

- Add support for some default conditions to start:
- `is_home`
- `is_single`
- `is_archive`
- `is_search`
- Determining if an item is the first in a list of items, such as in a Query Loop block, via a post template.

### PR DESCRIPTION
**Pull request by jakewrfoster**
## Summary

[wp-conditional-blocks] Basic Conditions add basic default conditions

## Notes for reviewers

None.

### Added
- a shortlist of "default" conditions
- a select menu in the slotfill sidebar for selecting conditions added via filters.

### Changed
- logic in render function callback to align with new condition attribute.

### Removed
- logic pertaining to the previous method of adding a condition (including, conditions, query, post, index)
- fields pertaining to the above from the slotfill sidebar

## Ticket(s)

- [TECH-84](https://alleyinteractive.atlassian.net/browse/TECH-84)

[TECH-84]: https://alleyinteractive.atlassian.net/browse/TECH-84?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: blocks/condition/edit.tsx [hunk 1]
```diff
@@ -48,90 +59,22 @@ export default function Edit({
...
-          </PanelRow>
-
-          {/* @ts-ignore-next-line */}
-          <PanelRow>
-            {/* @ts-ignore-next-line */}
-            <TextControl
-              label={__('Condition', 'wp-conditional-blocks')}
-              help={__('Any other condition', 'wp-conditional-blocks')}
+            <SelectControl
+              label={__('Conditionals', 'wp-conditional-blocks')}
```
Comment (scottnelle) [inline blocks/condition/edit.tsx]: Double checking that you want to use "Conditional" here and "Condition" in the help text. Should both be consistent?

File: src/class-conditions.php [hunk 2]
```diff
@@ -140,4 +149,16 @@ function ( $condition ) use ( $slug ) {
 	public function reset_conditions_for_testing(): void {
 		$this->conditions = [];
 	}
+
+	/**
+	 * Crates and adds default conditions.
```
Comment (scottnelle) [inline src/class-conditions.php]:
```suggestion
	 * Creates and adds default conditions.
```

**Review by scottnelle — APPROVED** [review]
👻

**Review by mogmarsh — APPROVED** [review]
looks good! 👍

**Pull request merged**